solidity/security/basic-arithmetic-underflow.yaml

```diff
@@ -19,12 +19,23 @@ rules:
     mode: taint
     pattern-sinks:
       - pattern: $Y - $X
+      - pattern: $Y -= $X
     pattern-sources:
-        - pattern-either:
-            - pattern-inside: |
-                function $F(..., $X, ...) external { ... }
-            - pattern-inside: |
-                function $F(..., $X, ...) public { ... }
+      - patterns:
+          - pattern-inside: |
+              function $F(..., $TYPE $X, ...) public { ... }
+          - focus-metavariable: $X
+      - patterns: 
+          - pattern-inside: |
+              function $F(..., $TYPE $X, ...) external { ... }
+          - focus-metavariable: $X
+      - patterns: 
+          - pattern-inside: |
+              function $F(..., $TYPE $X, ...) { ... }
```

Comment:
Yes, it matches the internal functions.
When I deleted it, semgrep stopped matching functions without visibility. This may be critical for contracts with pragma solidity prior to version 0.5.0, since visibility is public by default. This should be known when testing old contracts.